src/wp-admin/upgrade.php

```diff
@@ -54,6 +55,20 @@
 	$mysql_compat = version_compare( $mysql_version, $required_mysql_version, '>=' );
 }
 
+$missing_extensions = array();
+
+foreach ( $required_php_extensions as $extension ) {
+	if ( ! extension_loaded( $extension ) ) {
+		$missing_extensions[] = sprintf(
+			/* translators: 1: URL to WordPress release notes, 2: WordPress version number, 3: The PHP extension name needed. */
+			__( 'You cannot install because <a href="%1$s">WordPress %2$s</a> requires the %3$s PHP extension.' ),
```

Comment:
```suggestion
			__( 'You cannot upgrade because <a href="%1$s">WordPress %2$s</a> requires the %3$s PHP extension.' ),
```

I tested this by bumping the db version in `version.php` si it looks like the version is already running when this file is hit.